### ISSUE TITLE
Potential issue in TD3-BC BC loss calculation.

### ISSUE THREAD

**Issue by ethanluoyc** [deepmind/acme#219]
Hi,
I found a potential issue when using the TD3-BC agent in offline RL.
The question is about the following line.
https://github.com/deepmind/acme/blob/8fbc176ed4b31f869bc468bbbfbd1258ac8f324f/acme/agents/jax/td3/learning.py#L118

It seems that bc_loss is reduced with `jnp.mean` before adding to the dpg_loss, but the dpg_loss itself not reduced over the batch yet.
https://github.com/deepmind/acme/blob/8fbc176ed4b31f869bc468bbbfbd1258ac8f324f/acme/agents/jax/td3/learning.py#L114
Shouldn't this be
```python
dq_da = grad_critic(critic_params, transition.observation, action)
batch_dpg_learning = jax.vmap(rlax.dpg_loss, in_axes=(0, 0))
loss = jnp.mean(batch_dpg_learning(action, dq_da)) # Reduce over the batch.
if bc_alpha is not None:
  # BC regularization for offline RL
  bc_factor = jax.lax.stop_gradient(bc_alpha / jnp.mean(jnp.abs(dq_da)))
  loss += jnp.mean(jnp.square(action - transition.action)) / bc_factor
return loss # loss already reduced over the batch to dropping the jnp.mean.
```

**Comment by alexis-jacq** [comment]
Hi @ethanluoyc, thanks for the finding!

If I understand correctly, you suggest to replace `jnp.mean(loss + jnp.mean(bc_loss))` by `jnp.mean(loss) + jnp.mean(bc_loss)`.
It won't change the behaviour of the code (the mean operator being linear), but we may get a tiny gain in avoiding the broadcasting, so it is worth to make that change. Thanks!

**Comment by alexis-jacq** [comment]
Oh, that however is an actual bug:
I used dq_da as it was the critic, and actually it's the gradient of the critic loss. 
It indeed explains why I needed different values for alpha to make it work. 
It's interesting that it worked as well (reproducing similar results)!
Thanks a lot for your careful review, I will correct it!
